### PR TITLE
Add haystack index

### PR DESCRIPTION
**Pull request by yakky**

### CODE REVIEW

File: djangocms_blog/search_indexes.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+from django.utils.encoding import force_text
+from django.utils.text import force_unicode
```
Comment (yakky) [inline djangocms_blog/search_indexes.py]: Change with force_text

File: djangocms_blog/search_indexes.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+from aldryn_search.helpers import get_plugin_index_data
+from aldryn_search.utils import get_index_base, strip_tags
+from haystack import indexes
+
+from .models import Post
+from .settings import get_setting
+
+try:
+    from django.utils.encoding import force_text as force_unicode_or_text
```
Comment (yakky) [inline djangocms_blog/search_indexes.py]: this is not needed. Just use ``from django.utils.encoding import force_text`` it exists in every supported Django version

Comment (fmarco) [inline djangocms_blog/search_indexes.py]: :+1:

**Comment by yakky** [comment]
@fmarco I rebased on current develop with minimal changes to make it work. We need to check the desired behavior with multiple apphooks. Also: a test is failing apparently due to some leak between tests (``test_blog_post_is_indexed_using_prepare`` are still in the index in ``test_blog_post_is_indexed_using_update_object``). Apparently a index.clear() does not fix this

**Pull request merged**